### PR TITLE
[SPARK-37461][YARN][FOLLOWUP] Refactor YARN Client code to avoid add unnecessary parameter of `appId`

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
In https://github.com/apache/spark/pull/34710, we assign ApplicationId to `appId` in client mode too. After this change we can refactor more code:

1. We add a method `getApplicationId` to get `appId` from `Client`, and avoid it can be changed outside of `Client`.
2. `submitApplication()` don't return `appId` now. we need to call `getApplicationId` instead.
3. Remove `appId` argument from `monitorApplication()` and `getApplicationReport()`.

### Why are the changes needed?
Refactor code.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existed UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
@tgravescs Now may be what you want. cc @srowen

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -910,7 +913,7 @@ private[spark] class Client(
    * Set up a ContainerLaunchContext to launch our ApplicationMaster container.
    * This sets up the launch environment, java options, and the command for launching the AM.
    */
-  private def createContainerLaunchContext(newAppResponse: GetNewApplicationResponse)
+  private def createContainerLaunchContext()
     : ContainerLaunchContext = {
```
Comment (tgravescs) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala:917]: nit, can probably put this on previous line

**Review by tgravescs — APPROVED** [review]

**Review by mridulm — APPROVED** [review]
Looks good to me, will let @tgravescs take a pass.

File: resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala [hunk 2]
```diff
@@ -74,7 +74,7 @@ private[spark] abstract class YarnSchedulerBackend(
   private implicit val schedulerEndpointEC = ExecutionContext.fromExecutorService(
       ThreadUtils.newDaemonSingleThreadExecutor("yarn-scheduler-endpoint"))
 
-  /** Application ID. */
+  /** Application ID. This work for both cluster and client mode. */
```
Comment (mridulm) [inline resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala:77]: super nit: We dont really need to call this out

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: Done

File: resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala]: Done

**Comment by mridulm** [comment]
Merged to master.
Thanks for working on this @AngersZhuuuu !

Thanks for proposing and reviewing this @tgravescs :-)

**Pull request closed without merging**